flink-connector-bigquery-integration-test/src/main/java/com/google/cloud/flink/bigquery/integration/BigQueryIntegrationTest.java

```diff
@@ -0,0 +1,333 @@
+import org.apache.flink.util.Collector;
+
+import com.google.cloud.flink.bigquery.common.config.BigQueryConnectOptions;
+import com.google.cloud.flink.bigquery.source.BigQuerySource;
+import com.google.cloud.flink.bigquery.source.config.BigQueryReadOptions;
+import org.apache.avro.generic.GenericRecord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Duration;
+
+/**
+ * The Integration Test pipeline will try to read the specified BigQuery table according to the
+ * command line arguments, returning {@link GenericRecord} representing the rows, perform certain
+ * operations and then log the total number of records read. <br>
+ * This module checks the following cases of BigQuery Table read.
+ *
+ * <ol>
+ *   <li>Bounded Jobs: Involve reading a BigQuery Table in the <i> bounded </i> mode.<br>
+ *       The arguments given in this case would be:
+ *       <ul>
+ *         <li>--gcp-project {required; project ID which contains the BigQuery table}
+ *         <li>--bq-dataset {required; name of BigQuery dataset containing the desired table} <br>
+ *         <li>--bq-table {required; name of BigQuery table to read} <br>
+ *         <li>--agg-prop {required; record property to aggregate in Flink job} <br>
+ *         <li>--query {optional; SQL query to fetch data from BigQuery table}
+ *       </ul>
+ *       The sequence of operations in this pipeline is: <i>source > flatMap > keyBy > sum </i> <br>
+ *       A counter counts the total number of records read (the number of records observed by keyBy
+ *       operation) and logs this count at the end. <br>
+ *       Command to run bounded tests on Dataproc Cluster is: <br>
+ *       {@code gcloud dataproc jobs submit flink --id {JOB_ID} --jar= {GCS_JAR_LOCATION}
+ *       --cluster={CLUSTER_NAME} --region={REGION} -- --gcp-project {GCP_PROJECT_ID} --bq-dataset
+ *       {BigQuery Dataset Name} --bq-table {BigQuery Table Name} --agg-prop
+ *       {PROPERTY_TO_AGGREGATE_ON} --query {QUERY} } <br>
+ *       The following cases are tested:
+ *       <ol>
+ *         <li>Reading a Simple Table: This test reads a simple table of 40,000 rows having size 900
+ *             KBs.
+ *         <li>Reading a Table with Complex Schema: This test reads a table with 15 levels (maximum
+ *             number of levels allowed by BigQuery). The table contains 100,000 rows and has a size
+ *             of 2.96 MB.
+ *         <li>Reading a Large Table: This test reads a large table. The table contains __ rows and
+ *             has a size of about 200 GBs.
+ *         <li>Reading a Table with Large Row: This test reads a table with a large row. The table
+ *             contains 100 rows each fo size 45 MB and has a size of about 450 GB.
+ *         <li>Testing a BigQuery Query Run: This tests a BigQuery Query run. The query filters
+ *             certain rows based on a condition, groups the records and finds the AVG of value of a
+ *             column.
```

Comment:
This file is not aware of the tests being executed using it's main application. Please remove this part of the description. The nature of tests should be inferred from `nightly.yaml` and `nightly.sh`.

Reply:
Done.